app/models/miq_request_workflow.rb

```diff
@@ -33,6 +33,19 @@ def self.all_encrypted_options_fields
     descendants.flat_map(&:encrypted_options_fields).uniq
   end
 
+  def self.update_requester_from_parameters(data, user)
+    return user if data[:user_name].blank?
+    new_user = User.lookup_by_identity(data[:user_name])
+
+    unless new_user
+      _log.error "requested not changed to <#{data[:user_name]}> due to a lookup failure"
+      raise ActiveRecord::RecordNotFound
+    end
+
+    _log.warn "requested changed to <#{new_user.userid}>"
+    new_user
```

Comment:
@mkanoor I agree we will likely need those changes but I am suggesting that we do that in a separate PR along with the other work @tinaafitz needs.  I would like to keep this PR to just the `user_name` lookup.